server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -151,6 +152,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("ProcessQuery-"+queryId);
```

Comment:
At this point the code should be in a jetty thread, not a processing thread. Suggest `"qtp-query-" + queryId`